### PR TITLE
Exhaust existing poll request (eg. timed out queues)

### PR DESCRIPTION
**Pull request by cho45**
There is a problem around client timeouts. This patch fixes unnecessary queue accumulation.

(Timeouts may be caused by the front-end (authentication) http server)

### CODE REVIEW

**Comment by brian-brazil** [comment]
Can you explain more about what this is fixing?

**Comment by cho45** [comment]
Thank you for replying.

Reproducing problem step is following:
 
1. ./client connects to ./proxy via http frontend
 2. http frontend timeout the connection (prometheus does not request for a time)
 3. ./client receives timeout response
 4. ./client reconnects to ./proxy
 5. prometheus requests via ./proxy
 6. ./proxy receives request and send it ./client. But PushProx send it to old timed out ./client connection. So request will be failed.

Old connection (step 3) will accumulate in golang channel (from `c.getRequestChannel(fqdn)`). So PushProx will fail to request until exhausting old connections.

This patch clears old connections in golang channel when new ./client is connected (step 4).

File: proxy/proxy.go [hunk 1]
```diff
@@ -67,7 +67,12 @@ func main() {
 		// Client registering and asking for scrapes.
 		if r.URL.Path == "/poll" {
 			fqdn, _ := ioutil.ReadAll(r.Body)
-			request, _ := coordinator.WaitForScrapeInstruction(strings.TrimSpace(string(fqdn)))
+			request, err := coordinator.WaitForScrapeInstruction(strings.TrimSpace(string(fqdn)))
+			if err != nil {
+				level.Info(logger).Log("msg", "Request is expired")
```
Comment (brian-brazil) [inline proxy/proxy.go]: You should use the err here to get the message

**Comment by cho45** [comment]
Fixed the error messages following other parts with `err` object.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**